File: java/src/main/java/com/audi/interview/booking/jpa/domain/Vehicle.java
package com.audi.interview.booking.jpa.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Column;
import javax.persistence.Entity;
import java.time.LocalDate;

@Data
@Entity
@EqualsAndHashCode(callSuper = true)
public class Vehicle extends AbstractPersistable {

    @Column(unique = true)
    private String licensePlate;

    private String vin;

    private String model;

    private Boolean active;

    private String color;

    private LocalDate validTill;

}

File: node/src/vehicle/vehicle.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity({
  name: "vehicles"
})
export class Vehicle {
  @PrimaryGeneratedColumn("uuid")
  id: string;

  @Column({ unique: true })
  licensePlate: string;

  @Column()
  vin: string;

  @Column()
  model: string;

  @Column()
  active: boolean;

  @Column()
  color: string;

  @Column()
  validTill: Date;
}
